File: Queue.java
package com.company;

public class Queue<T> {

    private int _size;
    private Node<T> _head;

    public Queue(){
        _size = 0;
        _head = null;
    }

    public boolean isEmpty(){
        return _size == 0;
    }

    public void enqueue(T val){

        Node<T> newNode = new Node<T>(val);

        if(isEmpty())
        {
            this._head = newNode;
        }
        else
        {
            Node<T> last = this._head;
            while (last.get_next() != null)
                last = last.get_next();

            last.set_next(newNode);
        }
        this._size += 1;
    }

    public T dequeue(){

        if(isEmpty())
            return null;

        T info = this._head.get_info();
        this._head = this._head.get_next();
        this._size -= 1;
        return info;
    }

}

File: queue.py
class Node(object):

    def __init__(self, val):
        self.info = val
        self.next = None

class Queue(object):
    def __init__(self):
        self.size = 0
        self.head = None

    def isEmpty(self):
        return self.size == 0

    def enqueue(self, val):
        newNode = Node(val);

        if self.head is None:
            self.head = newNode
        else:
            last = self.head
            while last.next:
                last = last.next
            last.next = newNode
        self.size = self.size + 1

    def dequeue(self):

        if self.isEmpty():
            return None

        info = self.head.info
        self.head = self.head.next
        self.size = self.size - 1
        return info
